openssl/src/cipher_ctx.rs

```diff
@@ -591,6 +591,34 @@ impl CipherCtxRef {
         Ok(len)
     }
 
+    /// Like [`Self::cipher_update`] except that it writes output into the `data`
+    /// buffer. This function can only be used for stream ciphers.
+    /// Use [`Self::cipher_update`] with no output argument to write AAD.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the underlying cipher is not a stream cipher.
+    pub fn cipher_update_inplace(&mut self, data: &mut [u8]) -> Result<usize, ErrorStack> {
+        assert!(
+            self.block_size() == 1,
```

Comment:
Are you sure in place updates aren't supported for block ciphers? I'd expect you may just need to have an oversized buffer - something like `pub fn cipher_update_inplace(&mut self, buf: &mut [u8], in_bytes: usize) -> Result<usize, ErrorStack>` where `buf.len() > in_bytes + self.block_size()` or whatever.

Reply:
I hadn't considered that (extra parameter), makes much more sense. Left the name as `inlen` to match the openssl API.